src/app/new-auth/change-password/change-password.component.ts

```diff
@@ -1,15 +1,52 @@
+import { animate, style, transition, trigger } from '@angular/animations';
 import { Component, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { AuthValidationService } from 'app/core/auth-validation.service';
+import { AuthService } from 'app/core/auth.service';
+import { MatchValidator } from 'app/shared/validators/MatchValidator';
 
 @Component({
   selector: 'clark-change-password',
   templateUrl: './change-password.component.html',
-  styleUrls: ['./change-password.component.scss']
+  styleUrls: ['./change-password.component.scss'],
+  animations: [
+    trigger('fadeIn', [
+      transition(':enter', [
+        style({ opacity: 0 }),
+        animate('1000ms')
+      ])
+    ])
+  ]
 })
 export class ChangePasswordComponent implements OnInit {
+  errorMessage: String;
+  showError: Boolean;
+  otaCode: string;
+  done = false;
 
-  constructor() { }
+  passwords: FormGroup = new FormGroup({
+    'password': this.authValidationService.getInputFormControl('password'),
+    'confirmPassword': this.authValidationService.getInputFormControl('password')
+  }, { validators: MatchValidator.mustMatch('password', 'confirmPassword')});
+
+  constructor(private authValidationService: AuthValidationService,
+              private authService: AuthService,
+              private activatedRoute: ActivatedRoute) {
+  }
 
   ngOnInit(): void {
+    this.authValidationService.getErrorState().subscribe(err => this.showError = err);
+    this.otaCode = this.activatedRoute.queryParams['_value']['otaCode'];
   }
 
+  submit(): void {
+    this.authService.resetPassword(this.passwords.get('password').value, this.otaCode)
```

Comment:
Where does this function get called? The submit button doesn't seem to call this function.

Reply:
I think it's a form so when a form is submitted (i.e. the button is clicked even though the click listener isn't added) this function will run.